cpp/src/partitioning/partitioning.cu

```diff
@@ -681,28 +676,30 @@ struct dispatch_map_type {
       rmm::exec_policy(stream), histogram.begin(), histogram.end(), histogram.begin());
 
     // Copy offsets to host
-    std::vector<size_type> partition_offsets(histogram.size());
-    thrust::copy(histogram.begin(), histogram.end(), partition_offsets.begin());
+    auto const partition_offsets = cudf::detail::make_std_vector_sync(histogram, stream);
 
     // Unfortunately need to materialize the scatter map because
     // `detail::scatter` requires multiple passes through the iterator
-    rmm::device_vector<MapType> scatter_map(partition_map.size());
+    rmm::device_uvector<MapType> scatter_map(partition_map.size(), stream);
 
     // For each `partition_map[i]`, atomically increment the corresponding
     // partition offset to determine `i`s location in the output
     thrust::transform(rmm::exec_policy(stream),
                       partition_map.begin<MapType>(),
                       partition_map.end<MapType>(),
                       scatter_map.begin(),
-                      [offsets = histogram.data().get()] __device__(auto partition_number) {
+                      [offsets = histogram.data()] __device__(auto partition_number) {
                         return atomicAdd(&offsets[partition_number], 1);
                       });
 
     // Scatter the rows into their partitions
     auto scattered =
       cudf::detail::scatter(t, scatter_map.begin(), scatter_map.end(), t, false, stream, mr);
 
-    return std::make_pair(std::move(scattered), std::move(partition_offsets));
+    // Copy offsets to host and return
+    return std::make_pair(
+      std::move(scattered),
+      std::move(partition_offsets));  // cudf::detail::make_std_vector_sync(histogram, stream));
```

Comment:
The comment is incorrect, and I'm not sure what the purpose of the commented out code is. Was there a problem with creating the vector here?

Reply:
I never miss a chance to share this: https://nedbatchelder.com/text/deleting-code.html

Reply:
I accidentally left this in when I was debugging. Thanks for catching. I usually don't leave commented out code intentionally, and if I do I always leave a comment.